programs/survival/fakeGui.sc

```diff
@@ -0,0 +1,402 @@
+    global_fakeplayersscreen:fakeplayer = [screen, models];
+
+    for(models, if (_ != null,
+        inventory_set(screen, _:'slot', 1, _:'item' || 'minecraft:structure_void', nbt('{display:{Name:\'"'+_:'title'+'"\'},HideFlags:3}'))
+    ));
+    setAir(screen);
+);
+soundDell(player) -> sound('block.note_block.bell', player~'pos');
+setAir(screen, ...options) -> fromListSetAir(screen, range(options:0 || 9*3));
+fromListSetAir(screen, listIndex) -> 
+    for (listIndex, if (
+        inventory_get(screen, _) == null,
+        // I18n
+        inventory_set(screen, _, 1, 'minecraft:light_gray_stained_glass_pane', nbt('{display:{Name:\'""\'},HideFlags:3}'))
+    ));
+setScreenInventory(models, screen, fakeplayer) -> if (models, for (models, (
+    [playerSlot, screenSlot] = [_:0, _:1];
+    [item, count, nbt] = inventory_get(fakeplayer, playerSlot) || ['minecraft:air', 0, null];
+    inventory_set(screen, screenSlot, count, item, nbt);
+)));
+setPlayerInventory(models, screen, fakeplayer) -> if (models, for (models, (
+    [playerSlot, screenSlot] = [_:0, _:1];
+    [item, count, nbt] = inventory_get(screen, screenSlot) || ['minecraft:air', 0, null];
+    inventory_set(fakeplayer, playerSlot, count, item, nbt);
+)));
+setSelected(screen, fakeplayer, slot) -> (
+    for (range(9), inventory_set(screen, _, 1, 'minecraft:structure_void', null));
+    inventory_set(screen, slot, 1, 'minecraft:barrier', nbt('{Enchantments:[{id:void}]}'));
+    run('player ' + fakeplayer~'command_name' + ' hotbar ' + (number(slot)+1));
+);
+
+create_datapack('invupd', 
+    {
+        // I18n
+        'readme.txt' -> ['This datapack is built by carpet script','Original Author: scarlet', 'Remake: monkey'],
+        'data' -> {
+            'chyx' -> {
+                'advancements' -> {
+                    'xd.json '-> {
+                        'rewards' -> {'function' -> 'chyx:invupd'},
+                        'criteria' -> {
+                            'example' -> {
+                                'trigger' -> 'minecraft:inventory_changed',
+                                'conditions' -> {}
+                            }
+                        }
+                    }
+                },
+                'functions' -> {
+                    'invupd.mcfunction' -> 'script run signal_event(\'invupd\', null, player())\nadvancement revoke @s only chyx:xd'
+                }
+            }
+        }
+    }
+);
+handle_event('invupd', _(fakeplayer) -> (
+    screen = global_fakeplayersscreen:fakeplayer;
+    if(screen, setScreenInventory(screen:1, screen:0, fakeplayer));
+));
+__on_player_disconnects(fakeplayer, reason)->(
+    screen = global_fakeplayersscreen:fakeplayer;
+    if(screen, (
+        drop_item(screen:0, -1);
+        close_screen(screen:0);
+    ));
+);
+__on_player_interacts_with_entity(creativeplayer, fakeplayer, hand) -> if (
+    creativeplayer == fakeplayer || creativeplayer~'player_type' != 'singleplayer' || hand != 'mainhand', null,
```

Comment:
`creativeplayer~'player_type' != 'singleplayer'`

should it be `creativeplayer~'player_type' == 'fake'` ?